packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt

```diff
@@ -222,19 +231,28 @@ public class AuthCognito : FlutterPlugin, ActivityAware, MethodCallHandler, Plug
       }
   }
 
-  private fun onSignIn (@NonNull flutterResult: Result, @NonNull request: HashMap<String, *>) {
+  private fun onSignIn (@NonNull flutterResult: Result, @NonNull request: HashMap<String, *>){
+
+    runBlocking {
       try {
-        FlutterSignInRequest.validate(request)
-        var req = FlutterSignInRequest(request)
-        Amplify.Auth.signIn(
-                req.username,
-                req.password,
-                { result -> prepareSignInResult(flutterResult, result) },
-                { error -> errorHandler.handleAuthError(flutterResult, error)}
-        );
+        var session: AWSCognitoAuthSession? = null
+        try {
+          session = fetchAuthSessionCoroutine() as AWSCognitoAuthSession
+        } catch (e: Exception) {
+          print(e)
+        }
+        if (session == null || !session.isSignedIn || session.userPoolTokens.error is AuthException.SessionExpiredException) {
+          val result = signInCoroutine(request)
+          prepareSignInResult(flutterResult, result)
+        } else {
+          throw  FlutterInvalidStateException("There is already a user signed in.", "Sign out before calling sign in.")
```

Comment:
Would this result in same (or similar) exception in Dart as if this happened in iOS library?

Reply:
Yes. .invalidState is an error in iOS.  We created this exception in our code for clarity, but we are NOT making an invalidState exception in dart.